discovery.md

```diff
@@ -109,296 +123,218 @@ to execute some logic, which might lead to the occurrence of new events.
-  "datacontenttype": "application/json",
-  "dataschema": "http://schemas.example.com/download/com.example.storage.object.create.json"
-}
-```
-
-#### `producer` entity
-
-Once discovery narrows down to a specific event `type` that we want to subscribe
-to from a specific `service`, the appropriate `producer` entity can be
-retrieved by key. The key is a composite key of the `service` and `type` names.
-
-##### `producer` entity Attributes
-
-The `Service` entity is the main component of the discovery API. This section
-covers the structure of that entity and the next section describes the
-requirements for the query API.
+- Example:
+  - `storage.read`
 
-###### service
+###### protocols
 
-- Type: `String`
-- Description: Identifies the service of which the event is related. The
-  `service` string SHOULD be human readable and can identify a top level
-  producer system.
+- Type: `List` of `String`
+- Description: This field describes the different values that might be passed
+  in the `protocol` field of the CloudSubscriptions API. The protocols with
+  existing CloudEvents bindings are identified as AMQP, MQTT3, MQTT5, HTTP,
+  KAFKA, and NATS. An implementation MAY add support for further
+  protocols. All services MUST support at least one delivery protocol, and MAY
+  support additional protocols.
 - Constraints:
   - REQUIRED
-  - MUST be a non-empty string
-- Examples:
-  - "GitHub.com"
-  - "Awesome Cloud Storage"
-  - "com.example.microservices.userlogin"
-
-###### produceruri
-
-- Type: `URI`
-- Description: Absolute URI that provides a link back to the producer, or
-  documentation about the producer. This is for human consumption.
-- Constraints:
-  - OPTIONAL
-  - If present, MUST be a non-empty URI
+  - MUST be non-empty.
 - Examples:
-  - "http://cloud.example.com/docs/blobstorage"
+  - `[ "HTTP" ]`
+  - `[ "HTTP", "AMQP", "KAFKA" ]`
 
-###### type
+##### type
 
 - Type: `String`
-- Description: CloudEvents [`type`](https://github.com/cloudevents/spec/blob/master/spec.md#type)
+- Description: CloudEvents
+  [`type`](https://github.com/cloudevents/spec/blob/master/spec.md#type)
   attribute.
 - Constraints:
   - REQUIRED
   - MUST be a non-empty string, following the constraints as defined in the
     CloudEvents spec.
 - Examples:
-  - "com.github.pull.create"
-  - "com.example.object.delete.v2"
+  - `com.github.pull.create`
+  - `com.example.object.delete.v2`
 
-###### specversion
+##### description (Type)
 
-- Type: `String` per [RFC 2046](https://tools.ietf.org/html/rfc2046)
-- Description: CloudEvents [`specversion`](https://github.com/cloudevents/spec/blob/master/spec.md#specversion)
-  that will be used for events published for this producer, event type
-  combination.
+- Type: `String`
+- Description: Human readable description.
 - Constraints:
-  - REQUIRED
-  - MUST be a non-empty string
+  - OPTIONAL
+  - If present, MUST be a non-empty string
 
-###### datacontenttype
+##### datacontenttype
 
 - Type: `String`
```

Comment:
Same here as above for `specversion`, but the subscription API does not support it, yet.

Reply:
Would this apply to all of the `dataschema...` attributes?

Reply:
If so, I'll do this in a follow-on PR so we can get the bulk of this merged. ok?

Reply:
It's probably worth a separate discussion. 
+1 for a separate PR